WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListFragment.kt

```diff
@@ -123,14 +136,65 @@ class ProductListFragment :
 
         initAddProductFab(binding.addProductButton)
 
-        if (viewModel.isSearching()) {
-            binding.productsSearchTabView.isVisible = true
-            binding.productsSearchTabView.show(this, viewModel.isSkuSearch())
-        } else {
-            viewModel.reloadProductsFromDb(excludeProductId = pendingTrashProductId)
+        addSelectionTracker()
+
+        when {
+            viewModel.isSearching() -> {
+                binding.productsSearchTabView.isVisible = true
+                binding.productsSearchTabView.show(this, viewModel.isSkuSearch())
+            }
+            // Do nothing on selection mode
+            viewModel.isSelecting() -> {}
+            else -> {
+                viewModel.reloadProductsFromDb(excludeProductId = pendingTrashProductId)
+            }
         }
     }
 
+    private fun addSelectionTracker() {
+        tracker = SelectionTracker.Builder(
+            "productSelection", // a string to identity our selection in the context of this fragment
+            binding.productsRecycler, // the RecyclerView where we will apply the tracker
+            ProductSelectionItemKeyProvider(binding.productsRecycler), // the source of selection keys
+            DefaultProductListItemLookup(binding.productsRecycler), // the source of information about recycler items
+            StorageStrategy.createLongStorage() // strategy for type-safe storage of the selection state
+        ).withSelectionPredicate(selectionPredicate)
+            .build() // allows multiple items to be selected without any restriction
+
+        productAdapter.tracker = tracker
+
+        tracker?.addObserver(
+            object : SelectionTracker.SelectionObserver<Long>() {
+                override fun onSelectionChanged() {
+                    val selectionCount = tracker?.selection?.size() ?: 0
+                    if (selectionCount > 0 && actionMode == null) {
+                        viewModel.enterSelectionMode()
+                        actionMode = (requireActivity() as AppCompatActivity)
+                            .startSupportActionMode(this@ProductListFragment)
+                    }
+                    when (selectionCount) {
+                        0 -> {
+                            viewModel.exitSelectionMode()
+                            actionMode?.finish()
+                        }
+                        else -> {
+                            actionMode?.title = StringUtils.getQuantityString(
+                                context = requireContext(),
+                                quantity = selectionCount,
+                                default = R.string.product_selection_count,
+                                one = R.string.product_selection_count_single
+                            )
+                        }
+                    }
```

Comment:
⚙️ Can we move this logic to ViewModel? We should only pass information about selection change and a number of selected items to VM and then orchestrate changes from ViewModel.

```suggestion
                    val selectionCount = tracker?.selection?.size() ?: 0
                    viewModel.onSelectionChanged(selectionCount)
```